packages/vite/src/node/ssr/ssrExternal.ts

```diff
@@ -179,8 +179,25 @@ function collectExternals(
 
 export function shouldExternalizeForSSR(
   id: string,
-  externals: string[]
+  externals: string[],
+  noExternal: string | RegExp | (string | RegExp)[] | true
 ): boolean {
+  if (noExternal === true) {
+    return false
+  }
+  const noExternalsArr = Array.isArray(noExternal) ? noExternal : [noExternal]
+  for (const e of noExternalsArr) {
+    if (typeof e === 'string') {
+      if (e === 'id') {
+        return false
+      } else {
+        continue
+      }
+    }
+    if (e.test(id)) {
+      return false
+    }
```

Comment:
You could write this more succinctly as:
```suggestion
    if (e === 'id') {
      return false
    } else if (typeof e === 'string') {
        continue
    } else if (e.test(id)) {
      return false
    }
```

Reply:
It looks great